src/Compiler/Driver/ParseAndCheckInputs.fsi

```diff
@@ -239,7 +255,7 @@ val CheckClosedInputSet:
     tcState: TcState *
     eagerFormat: (PhasedDiagnostic -> PhasedDiagnostic) *
     inputs: ParsedInput list ->
-        TcState * TopAttribs * CheckedImplFile list * TcEnv
+        TcState * TopAttribs * CheckedImplFile list * TcEnv * TcState list
```

Comment:
Also note, TcState list includes the first state. I just kept the first state separately here to decrease the change surface a bit for now.